integration/spark/app/src/main/java/io/openlineage/spark/agent/ArgumentParser.java

```diff
@@ -134,6 +160,83 @@ private static void extractSparkSpecificConfigEntriesFromSparkConf(
         .ifPresent(a -> config.getFacetsConfig().setDisabledFacets(a));
   }
 
+  private static Optional<SparkOpenLineageConfig> extractOpenLineageConfFromEnvVar() {
```

Comment:
Would it make sense to create separate class `EnvPropertiesArgumentParser` with newly created methods?

Reply:
I would take a slightly different approach. I would use `io.openlineage.spark.agent.ArgumentParser#extractOpenLineageConfFromSparkConf` as a base (with changed name, for example `configFromProps`). Instead of `SparkConf`, I would accept `List<Pair> props`. Then, in `io.openlineage.spark.agent.ArgumentParser#extractOpenLineageConfFromSparkConf` I would call `filterProperties(conf)` and pass it to the new method. Similarly, I would create extractLineageConfFromEnvVariables(), where I would use environmental variables to create a list of pairs ready to pass to the new extracted method. What do you think?